website/source/docs/job-specification/affinity.html.md

```diff
@@ -28,8 +28,38 @@ be expressed on [attributes][interpolation] or [client metadata][client-meta].
 Additionally affinities may be specified at the [job][job], [group][group], or
 [task][task] levels for ultimate flexibility.
 
-Placing the same affinity at both the job level and at the group level is redundant
-since affinities are applied hierarchically. The job affinities will affect
+```hcl
+job "docs" {
+  # Prefer nodes in the us-west1 datacenter
+  affinity {
+    attribute = "${node.datacenter}"
+    value     = "us-west1"
+    weight    = 100
+  }
+
+  group "example" {
+    # Prefer the "r1" rack
+    affinity {
+      operator  = "${meta.rack}"
+      value     = "r1"
+      weight    = 50
+    }
+
+    task "server" {
+      # Prefer nodes where "my_custom_value" is greater than 5
+      affinity {
+        attribute = "${meta.my_custom_value}"
+        operator  = ">"
+        value     = "3"
+        weight    = 50
+      }
+    }
+  }
+}
+```
+
+Placing affinities at both the job level and at the group level is redundant
+since constraints are applied hierarchically. The job constraints will affect
```

Comment:
This sentence is hard for me to parse. The only "redundant" part is that there's only 1 group and 1 task, so there's no reason to split the affinities across all 3 stanzas? Although even if you combined them all within the group you'd still need 3 affinities stanzas, so I'm not sure what "redundant" refers to.

Also "constraints are applied hierarchically" should probably be "affinities are applied hierarchically" since they're distinct stanzas and we're specifically discussing affinities here.

I'm not even sure why we allow affinities on tasks since affinities are always applied to the group right? So it's not a hierarchy that flows down job -> group -> task. Job and task affinities just get applied to groups?

So perhaps this would clear it all up:

> Affinities apply to task groups but may be specified within job and task stanzas as well. Job affinities apply to all groups within the job. Task affinities applies to the whole group the task is a part of.

Reply:
bad copy paste bug here, will update it with your suggestion it looks much nicer